websites/Y/YouTube Music/presence.ts

```diff
@@ -2,90 +2,263 @@ const presence = new Presence({
+			videoElement.addEventListener("play", videoListener);
+
+			videoListenerAttached = true;
+		}
+		//* Element got removed from the DOM (eg, song with song/video switch)
+	} else {
+		prevTitleAuthor = "";
+		videoListenerAttached = false;
+	}
+
+	presenceData = null;
+
+	if (hidePaused && mediaSession.playbackState !== "playing")
+		return presence.clearActivity();
+
+	if (["playing", "paused"].includes(mediaSession.playbackState)) {
+		if (privacyMode) {
+			return mediaSession.playbackState === "playing"
+				? presence.setActivity({
+						largeImageKey: "ytm_lg",
+						details: "Listening to music",
+				  })
+				: presence.setActivity();
+		}
+
+		if (
+			!mediaSession.metadata?.title ||
+			//* Check if the video has a duration
+			isNaN(videoElement.duration)
+		)
+			return;
+
+		if (
+			prevTitleAuthor !==
+			mediaSession.metadata.title + mediaSession.metadata.artist
+		) {
+			mediaTimestamps = presence.getTimestampsfromMedia(videoElement);
+			//* Don't ask me why the above function doesn't floor the end timestamp
+			mediaTimestamps[1] = Math.floor(mediaTimestamps[1]);
+
+			prevTitleAuthor =
+				mediaSession.metadata.title + mediaSession.metadata.artist;
+		}
+
+		const albumArtistBtnLink = mediaSession.metadata.album
+				? [...document.querySelectorAll<HTMLAnchorElement>(".byline a")].at(-1)
+						?.href
+				: document.querySelector<HTMLAnchorElement>(".byline a")?.href,
+			buttons: [ButtonData, ButtonData?] = [
+				{
+					label: "Listen Along",
+					url: `https://music.youtube.com/watch?v=${watchID}`,
+				},
+			];
+
+		if (albumArtistBtnLink) {
+			buttons.push({
+				label: `View ${mediaSession.metadata.album ? "Album" : "Artist"}`,
+				url: albumArtistBtnLink,
+			});
+		}
+
+		presenceData = {
+			largeImageKey: showCover
+				? mediaSession.metadata.artwork.at(-1).src
+				: "ytm_lg",
+			details: mediaSession.metadata.title,
+			state: [mediaSession.metadata.artist, mediaSession.metadata.album]
+				.filter(Boolean)
+				.join(" - "),
+			...(showButtons && {
+				buttons,
+			}),
+			smallImageKey:
+				mediaSession.playbackState === "paused"
 					? "pause"
 					: repeatMode === "ONE"
 					? "repeat-one"
 					: repeatMode === "ALL"
 					? "repeat"
 					: "play",
-				smallImageText: video.paused
+			smallImageText:
+				mediaSession.playbackState === "paused"
 					? "Paused"
 					: repeatMode === "ONE"
 					? "On loop"
 					: repeatMode === "ALL"
 					? "Playlist on loop"
 					: "Playing",
-				startTimestamp: prevMetadata?.startedAt,
-				endTimestamp,
-			};
+			...(showTimestamps &&
+				mediaSession.playbackState === "playing" && {
+					startTimestamp: mediaTimestamps[0],
+					endTimestamp: mediaTimestamps[1],
+				}),
+		};
+	} else if (showBrowsing) {
+		if (privacyMode) {
+			return presence.setActivity({
+				largeImageKey: "ytm_lg",
+				details: "Browsing YouTube Music",
+			});
+		}
+
+		if (oldPath !== document.location.pathname) {
```

Comment:
`document.location.pathname` should be stored as a destructured variable instead

Reply:
Don't resolve if you haven't resolved the conversation

Reply:
I resolved it tho?

Reply:
Not on all references